realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmResultsExtensions.kt

```diff
@@ -81,7 +82,9 @@ fun <T : RealmModel> RealmResults<T>.toFlow(): Flow<RealmResults<T>> {
         // Get instance to ensure the Realm is open for as long as we are listening
         val flowRealm = Realm.getInstance(config)
         val listener = RealmChangeListener<RealmResults<T>> { listenerResults ->
-            offer(listenerResults.freeze())
+            if (isActive) {
```

Comment:
It could very well be that the job running the flow is cancelled and the listener gets triggered after that happens. Nicely spotted!

Unfortunately, we have moved the logic away from the extension functions to a better suiting factory users will have the possibility of implementing to override the default behaviour just as we do with `RxObservableFactory`, so this change is no longer valid in the current `RealmResultsExtensions`.

I will see to add it in the factory. As I see it, this fix will also be needed in the other `toFlow` functions.